File: evals/fixtures/flask-py/app/services/reservation_service.py
from models import store
from models.reservation import Guest, Reservation


def _reservation_to_dict(r: Reservation) -> dict:
    return {
        "id": r.id,
        "guest_id": r.guest_id,
        "unit_id": r.unit_id,
        "check_in": r.check_in,
        "check_out": r.check_out,
        "status": r.status,
    }


def create_reservation(data: dict) -> dict:
    unit_id = data.get("unit_id")
    unit = store.find_unit(unit_id)
    if not unit:
        raise ValueError(f"Unit {unit_id} not found")

    guest_email = data.get("guest_email")
    guest_name = data.get("guest_name")

    # Find existing guest by email or create new one
    guest = next((g for g in store.guests if g.email == guest_email), None)
    if not guest:
        guest = Guest(id=store.generate_id(), name=guest_name, email=guest_email)
        store.guests.append(guest)

    reservation = Reservation(
        id=store.generate_id(),
        guest_id=guest.id,
        unit_id=unit_id,
        check_in=data.get("check_in"),
        check_out=data.get("check_out"),
        status="confirmed",
    )
    store.reservations.append(reservation)

    return _reservation_to_dict(reservation)


def update_reservation(reservation_id: str, data: dict) -> dict:
    reservation = next(
        (r for r in store.reservations if r.id == reservation_id), None
    )
    if not reservation:
        raise LookupError(f"Reservation {reservation_id} not found")

    if "check_in" in data:
        reservation.check_in = data["check_in"]
    if "check_out" in data:
        reservation.check_out = data["check_out"]

    return _reservation_to_dict(reservation)


def cancel_reservation(reservation_id: str) -> dict:
    reservation = next(
        (r for r in store.reservations if r.id == reservation_id), None
    )
    if not reservation:
        raise LookupError(f"Reservation {reservation_id} not found")

    reservation.status = "cancelled"

    return _reservation_to_dict(reservation)


def get_reservation(reservation_id: str) -> dict:
    reservation = next(
        (r for r in store.reservations if r.id == reservation_id), None
    )
    if not reservation:
        raise LookupError(f"Reservation {reservation_id} not found")

    return _reservation_to_dict(reservation)

File: evals/fixtures/nextjs-ts/app/src/lib/services/reservationService.ts
import { Reservation } from "../models/types";
import { store, generateId, findUnit } from "../models/store";

export function createReservation(data: {
  guestName: string;
  guestEmail: string;
  unitId: string;
  checkIn: string;
  checkOut: string;
}): Reservation {
  const unit = findUnit(data.unitId);
  if (!unit) {
    throw new Error(`Unit not found: ${data.unitId}`);
  }

  // Find existing guest by email or create a new one
  let guest = store.guests.find((g) => g.email === data.guestEmail);
  if (!guest) {
    guest = {
      id: generateId(),
      name: data.guestName,
      email: data.guestEmail,
    };
    store.guests.push(guest);
  }

  const reservation: Reservation = {
    id: generateId(),
    guestId: guest.id,
    unitId: data.unitId,
    checkIn: data.checkIn,
    checkOut: data.checkOut,
    status: "confirmed",
  };

  store.reservations.push(reservation);
  return reservation;
}

export function updateReservation(
  id: string,
  data: { checkIn?: string; checkOut?: string }
): Reservation {
  const reservation = store.reservations.find((r) => r.id === id);
  if (!reservation) {
    throw new Error(`Reservation not found: ${id}`);
  }

  if (data.checkIn !== undefined) {
    reservation.checkIn = data.checkIn;
  }
  if (data.checkOut !== undefined) {
    reservation.checkOut = data.checkOut;
  }

  return reservation;
}

export function cancelReservation(id: string): Reservation {
  const reservation = store.reservations.find((r) => r.id === id);
  if (!reservation) {
    throw new Error(`Reservation not found: ${id}`);
  }

  reservation.status = "cancelled";
  return reservation;
}

export function getReservation(id: string): Reservation {
  const reservation = store.reservations.find((r) => r.id === id);
  if (!reservation) {
    throw new Error(`Reservation not found: ${id}`);
  }

  return reservation;
}
